example/src/BasicPagerViewExample.tsx

```diff
@@ -1,46 +1,44 @@
-import React, { useMemo } from 'react';
-import { Image, StyleSheet, View, SafeAreaView, Animated } from 'react-native';
+import React from 'react';
+import { Image, SafeAreaView, StyleSheet, View } from 'react-native';
 
-import PagerView from 'react-native-pager-view';
+import { PagerView } from 'react-native-pager-view';
 
 import { LikeCount } from './component/LikeCount';
 import { NavigationPanel } from './component/NavigationPanel';
 import { useNavigationPanel } from './hook/useNavigationPanel';
+import { CreatePage, createPageKeyExtractor } from './utils';
 
-const AnimatedPagerView = Animated.createAnimatedComponent(PagerView);
+function renderItem({ item }: { item: CreatePage }) {
+  return (
+    <View style={item.style}>
+      <Image style={styles.image} source={item.imgSource} />
+      <LikeCount />
+    </View>
+  );
+}
 
 export function BasicPagerViewExample() {
-  const { ref, ...navigationPanel } = useNavigationPanel();
+  const { ref, ...navigationPanel } = useNavigationPanel<CreatePage>();
 
   return (
     <SafeAreaView style={styles.container}>
-      <AnimatedPagerView
+      <PagerView
         ref={ref}
-        style={styles.PagerView}
-        initialPage={0}
-        overdrag={navigationPanel.overdragEnabled}
+        style={styles.pagerView}
         scrollEnabled={navigationPanel.scrollEnabled}
         onPageScroll={navigationPanel.onPageScroll}
         onPageSelected={navigationPanel.onPageSelected}
         onPageScrollStateChanged={navigationPanel.onPageScrollStateChanged}
         pageMargin={10}
-        // Lib does not support dynamically orientation change
         orientation="horizontal"
-        // Lib does not support dynamically transitionStyle change
```

Comment:
please revert it

Reply:
In this branch it does work (iOS only; Android does not support any transition styles).

Reply:
Do you mean fast refresh ? bc if you pass a state here, it would not work.

Reply:
No, works on state change.  Can test by hacking the basic example; enable/disable dots will switch between horizontal/vertical, overdrag will switch between scroll/curl.

```diff
diff --git a/example/src/BasicPagerViewExample.tsx b/example/src/BasicPagerViewExample.tsx
index 99d49f0..295a78d 100644
--- a/example/src/BasicPagerViewExample.tsx
+++ b/example/src/BasicPagerViewExample.tsx
@@ -30,9 +30,9 @@ export function BasicPagerViewExample() {
         onPageSelected={navigationPanel.onPageSelected}
         onPageScrollStateChanged={navigationPanel.onPageScrollStateChanged}
         pageMargin={10}
-        orientation="horizontal"
+        orientation={navigationPanel.dotsEnabled ? 'horizontal' : 'vertical'}
         overdrag={navigationPanel.overdragEnabled}
-        transitionStyle="scroll"
+        transitionStyle={navigationPanel.overdragEnabled ? 'scroll' : 'curl'}
         showPageIndicator={navigationPanel.dotsEnabled}
         data={navigationPanel.pages}
         keyExtractor={createPageKeyExtractor}
```